### PR TITLE
[CRAW-143] Move fast/slow filtering to Generator

### PR DESCRIPTION
**Pull request by vanchpuck**

### CODE REVIEW

File: src/java/org/apache/nutch/fetcher/FetcherJob.java [hunk 1]
```diff
@@ -269,10 +250,10 @@ public FetcherJob(Configuration conf) {
   private FilterList getFilterList(String batchId, boolean useFastSlow) {    
     List<MapFieldValueFilter<String, WebPage>> filters = new ArrayList<>();
     filters.add(getBatchIdFilter(batchId));
-    if(useFastSlow) {
-      boolean isSlow = getConf().getBoolean(IS_SLOW_KEY, false);    
-      filters.add(getFastSlowFilter(isSlow));
-    }
+//    if(useFastSlow) {
```
Comment (SergeyWeiss) [inline src/java/org/apache/nutch/fetcher/FetcherJob.java]: I don't really think we need these comments here

File: src/java/org/apache/nutch/crawl/GeneratorMapper.java [hunk 2]
```diff
@@ -97,6 +99,19 @@ public void map(String reversedUrl, WebPage page,
...
+      boolean isSlowPage = markValue == null ? false : Bytes.toBoolean(markValue.getBytes());
+      boolean isSlowCrawler = context.getConfiguration().getBoolean(IS_SLOW_KEY, false);
+      if (isSlowCrawler != isSlowPage) {
+        if (GeneratorJob.LOG.isDebugEnabled()) {
+          GeneratorJob.LOG.debug("Skipping " + url + "; unsuitable page velocity category");
+        }
+        return;
+      }
+    }
+
```
Comment (SergeyWeiss) [inline src/java/org/apache/nutch/crawl/GeneratorMapper.java]: I think we'd better move this block to function with descriptive name, WDYT?

File: src/test/org/apache/nutch/util/CrawlTestUtilExt.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    public boolean isSlow() {
+      return isSlow;
+    }
+  }
+  
+  public static Configuration createConfiguration(Crawler crawlerType) {
+    Configuration conf = CrawlTestUtil.createConfiguration();
+    conf.setBoolean("fetcher.is.slow", crawlerType.isSlow());
+    conf.setStrings(CommonConfigurationKeys.IO_SERIALIZATIONS_KEY,
+        new String[] { PersistentSerialization.class.getName(),
```
Comment (SergeyWeiss) [inline src/test/org/apache/nutch/util/CrawlTestUtilExt.java]: It seems like array creation is redundant here

File: src/test/org/apache/nutch/crawl/TestGeneratorMapper.java [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    SelectorEntry entry = new SelectorEntry();
+    entry.set(URL, SCORE);
+    
+    MapDriver<String, WebPage, SelectorEntry, WebPage> driver = new MapDriver<String, WebPage, SelectorEntry, WebPage>();
+    driver.setConfiguration(conf);
+    
+    List<Pair<SelectorEntry, WebPage>> list = driver
+      .withMapper(new GeneratorMapper())
+      .withInput(INVERSED_URL, samplePage)
+      .withOutput(entry, samplePage)
```
Comment (SergeyWeiss) [inline src/test/org/apache/nutch/crawl/TestGeneratorMapper.java]: To my experience, withOutput is not needed

File: src/java/org/apache/nutch/crawl/GeneratorMapper.java (hunk 2 above)
Comment (vanchpuck) [inline src/java/org/apache/nutch/crawl/GeneratorMapper.java]: I just followed the code style, presented in this class. I'll create separate function.

File: src/test/org/apache/nutch/util/CrawlTestUtilExt.java (hunk 3 above)
Comment (vanchpuck) [inline src/test/org/apache/nutch/util/CrawlTestUtilExt.java]: Most likely, I copied this piece of code from gora. I'm not sure, that use of only one serialization class is reliable in this case.

Comment (SergeyWeiss) [inline src/test/org/apache/nutch/util/CrawlTestUtilExt.java]: No, I meant you don't have to write new String[] {...}, because setStrings is varargs method

**Comment by SergeyWeiss** [comment]
:+1:

**Pull request merged**